SearchServiceTests#testCreateSearchContextFailure

**Build scan**:

https://gradle-enterprise.elastic.co/s/mb4fxdioop3sk

**Repro line**:

```
./gradlew ':server:test' --tests "org.elasticsearch.search.SearchServiceTests" -Dtests.seed=3513A4B213D3E47A -Dtests.security.manager=true -Dtests.locale=ko -Dtests.timezone=Africa/Niamey -Druntime.java=11
```

**Reproduces locally?**:

No

**Applicable branches**:

master

**Failure history**:

I haven't found any other occurrence. 

**Failure excerpt**:

```
java.lang.AssertionError: should have 2 store refs (IndexService + InternalEngine) expected:<2> but was:<4>
        at __randomizedtesting.SeedInfo.seed([3513A4B213D3E47A:D9C6D58E7791BD50]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
```

--- a/server/src/test/java/org/elasticsearch/search/SearchServiceTests.java
+++ b/server/src/test/java/org/elasticsearch/search/SearchServiceTests.java
@@ -891,7 +891,7 @@ public class SearchServiceTests extends ESSingleNodeTestCase {
      * While we have no NPE in DefaultContext constructor anymore, we still want to guard against it (or other failures) in the future to
      * avoid leaking searchers.
      */
-    public void testCreateSearchContextFailure() throws IOException {
+    public void testCreateSearchContextFailure() throws Exception {
         final String index = randomAlphaOfLengthBetween(5, 10).toLowerCase(Locale.ROOT);
         final IndexService indexService = createIndex(index);
         final SearchService service = getInstanceFromNode(SearchService.class);
@@ -908,7 +908,9 @@ public class SearchServiceTests extends ESSingleNodeTestCase {
                 () -> service.createContext(reader, request, null, randomBoolean()));
             assertEquals("expected", e.getMessage());
         }
-        assertEquals("should have 2 store refs (IndexService + InternalEngine)", 2, indexService.getShard(0).store().refCount());
+        // Needs to busily assert because Engine#refreshNeeded can increase the refCount.
+        assertBusy(() ->
+            assertEquals("should have 2 store refs (IndexService + InternalEngine)", 2, indexService.getShard(0).store().refCount()));
     }
 
     public void testMatchNoDocsEmptyResponse() throws InterruptedException {

